src/Microsoft.Health.SqlServer/Features/Client/RetrySqlCommandWrapper.cs

```diff
@@ -8,74 +8,56 @@
 using System.Threading.Tasks;
 using EnsureThat;
 using Microsoft.Data.SqlClient;
-using Polly;
 
 namespace Microsoft.Health.SqlServer.Features.Client
 {
     /// <summary>
     /// A wrapper around <see cref="SqlCommand"/> to provide automatic retries for transient errors.
     /// </summary>
-    internal class RetrySqlCommandWrapper : SqlCommandWrapper
+    public class RetrySqlCommandWrapper : SqlCommandWrapper
     {
-        private readonly SqlCommandWrapper _sqlCommandWrapper;
-        private readonly IAsyncPolicy _retryPolicy;
-
-        public RetrySqlCommandWrapper(SqlCommandWrapper sqlCommandWrapper, IAsyncPolicy retryPolicy)
-            : base(sqlCommandWrapper)
+        public RetrySqlCommandWrapper(SqlCommand sqlCommand, SqlRetryLogicBaseProvider sqlRetryLogicBaseProvider)
+            : base(sqlCommand)
         {
-            EnsureArg.IsNotNull(sqlCommandWrapper, nameof(sqlCommandWrapper));
-            EnsureArg.IsNotNull(retryPolicy, nameof(retryPolicy));
+            EnsureArg.IsNotNull(sqlRetryLogicBaseProvider, nameof(sqlRetryLogicBaseProvider));
 
-            _sqlCommandWrapper = sqlCommandWrapper;
-            _retryPolicy = retryPolicy;
+            RetryLogicProvider = sqlRetryLogicBaseProvider;
         }
 
         /// <inheritdoc/>
-        public override Task<int> ExecuteNonQueryAsync(CancellationToken cancellationToken)
+        public override async Task<int> ExecuteNonQueryAsync(CancellationToken cancellationToken)
         {
-            return _retryPolicy.ExecuteAsync(async () =>
-            {
-                await EnsureConnectionOpenAsync(cancellationToken);
-                return await _sqlCommandWrapper.ExecuteNonQueryAsync(cancellationToken);
-            });
+           await EnsureConnectionOpenAsync(cancellationToken);
```

Comment:
Do we still need this here?

Reply:
If we don't, then we could probably also delete this class and simply assign the `SqlRetryLogicBaseProvider` inside of `SqlConnectionWrapper` when generating a retryable command